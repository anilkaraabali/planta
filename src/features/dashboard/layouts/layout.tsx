import { Logo } from '@/components/logo';
import { Button } from '@heroui/react';
import NextLink from 'next/link';
import { useRouter } from 'next/router';
import { FC, PropsWithChildren, useMemo } from 'react';
import {
  LiaChartBarSolid,
  LiaHeart,
  LiaHomeSolid,
  LiaSearchSolid,
} from 'react-icons/lia';
import { PiPlantLight } from 'react-icons/pi';

const DashboardLayout: FC<PropsWithChildren> = ({ children }) => {
  const router = useRouter();

  const menuItems = useMemo(
    () => [
      { href: '/dashboard', icon: LiaHomeSolid, name: 'Home' },
      { href: '/dashboard/search', icon: LiaSearchSolid, name: 'Plants' },
      { href: '/dashboard/me', icon: PiPlantLight, name: 'My list' },
      { href: '/dashboard/stats', icon: LiaChartBarSolid, name: 'Stats' },
      { href: '/favorites', icon: LiaHeart, name: 'Favorites' },
    ],
    []
  );

  return (
    <>
      <header className='flex h-16 items-center justify-between border-b border-gray-100 px-4'>
        <Logo />
      </header>
      <aside className='fixed bottom-0 left-0 top-16 z-10 flex w-16 border-r border-gray-100'>
        <nav className='flex size-full flex-col items-center justify-start space-y-4 py-4'>
          {menuItems.map((item, index) => (
            <Button
              as={NextLink}
              color='primary'
              href={item.href}
              isIconOnly
              key={index}
              variant={router.pathname === item.href ? 'flat' : 'light'}
            >
              <item.icon size={24} />
            </Button>
          ))}
        </nav>
      </aside>
      <main className='ml-16 h-[calc(100vh-64px)] overflow-auto bg-white p-4'>
        <div className='space-y-4'>{children}</div>
      </main>
    </>
  );
};

export { DashboardLayout };
