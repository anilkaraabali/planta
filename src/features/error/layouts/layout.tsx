import { DynamicTranslationKey } from '@/types';
import { Button } from '@heroui/react';
import { ErrorProps as NextErrorProps } from 'next/error';
import Image from 'next/image';
import NextLink from 'next/link';
import { AbstractIntlMessages, useTranslations } from 'next-intl';
import { FC } from 'react';
import { LiaArrowRightSolid } from 'react-icons/lia';

interface ErrorLayoutProps extends NextErrorProps {
  detail?: string;
  messages: AbstractIntlMessages;
}

const ErrorLayout: FC<ErrorLayoutProps> = (props) => {
  const t = useTranslations('Error');

  return (
    <div className='pt-24'>
      <div className='container flex flex-col items-center justify-center text-center'>
        <Image
          alt='Error illustration'
          className='mb-6'
          height={196}
          src='/illustrations/error.svg'
          width={396}
        />
        <h1 className='text-2xl' data-testid='error-title'>
          {t(`${props.statusCode}.title` as DynamicTranslationKey)}
        </h1>
        <p className='mt-2 text-default-500' data-testid='error-detail'>
          {t(`${props.statusCode}.detail` as DynamicTranslationKey)}
        </p>
        <Button
          as={NextLink}
          className='mt-6'
          color='primary'
          data-testid='error-cta'
          endContent={<LiaArrowRightSolid size={20} />}
          href='/'
          variant='shadow'
        >
          {t('cta')}
        </Button>
      </div>
    </div>
  );
};

export type { ErrorLayoutProps };
export { ErrorLayout };
