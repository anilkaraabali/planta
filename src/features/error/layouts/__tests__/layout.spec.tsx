import { render, screen } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import { ErrorLayout } from '../layout';

describe('ErrorLayout', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <ErrorLayout messages={{}} statusCode={500} />
    );

    expect(container).toMatchSnapshot();
  });

  test('should render correct title & detail for 404 error', () => {
    render(<ErrorLayout messages={{}} statusCode={404} />);

    expect(screen.getByTestId('error-title')).toHaveTextContent('404.title');
    expect(screen.getByTestId('error-detail')).toHaveTextContent('404.detail');
  });

  test('should render correct title & detail for 500 error', () => {
    render(<ErrorLayout messages={{}} statusCode={500} />);

    expect(screen.getByTestId('error-title')).toHaveTextContent('500.title');
    expect(screen.getByTestId('error-detail')).toHaveTextContent('500.detail');
  });

  test('should use fallback translation if statusCode is missing', () => {
    render(<ErrorLayout messages={{}} statusCode={400} />);

    expect(screen.getByTestId('error-title')).toHaveTextContent('400.title');
    expect(screen.getByTestId('error-detail')).toHaveTextContent('400.detail');
  });

  test('should render the CTA button correctly', async () => {
    render(<ErrorLayout messages={{}} statusCode={404} />);

    const button = screen.getByTestId('error-cta');

    expect(button).toHaveAttribute('href', '/');
  });
});
