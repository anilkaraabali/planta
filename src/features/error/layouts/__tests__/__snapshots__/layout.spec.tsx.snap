// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorLayout > should match snapshot 1`] = `
<div>
  <div
    class="pt-24"
  >
    <div
      class="container flex flex-col items-center justify-center text-center"
    >
      <img
        alt="Error illustration"
        class="mb-6"
        data-nimg="1"
        decoding="async"
        height="196"
        loading="lazy"
        src="/illustrations/error.svg"
        style="color: transparent;"
        width="396"
      />
      <h1
        class="text-2xl"
        data-testid="error-title"
      >
        500.title
      </h1>
      <p
        class="mt-2 text-default-500"
        data-testid="error-detail"
      >
        500.detail
      </p>
      <a
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none shadow-lg shadow-primary/40 bg-primary text-primary-foreground data-[hover=true]:opacity-hover mt-6"
        data-testid="error-cta"
        href="/"
        role="button"
        tabindex="0"
      >
        cta
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          tabindex="-1"
          viewBox="0 0 32 32"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 18.71875 6.78125 L 17.28125 8.21875 L 24.0625 15 L 4 15 L 4 17 L 24.0625 17 L 17.28125 23.78125 L 18.71875 25.21875 L 27.21875 16.71875 L 27.90625 16 L 27.21875 15.28125 Z"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
